R/as_rtf_pageby.R

```diff
@@ -139,10 +139,18 @@ as_rtf_pageby <- function(tbl) {
 
   ## Remove repeated records if group_by is not null
   if (!is.null(group_by)) {
-    cell_tbl <- rtf_group_by_enhance(cell_tbl,
-      group_by = group_by,
+    tmp <- cell_tbl
+    tmp$`.pageby` <- pageby$id
+    tmp$`.order` <- 1:nrow(tmp)
+    tmp <- rtf_group_by_enhance(tmp,
+      group_by = c(".pageby", group_by),
       page_index = subset(page_dict, !pageby)$page
     )
+
+    stopifnot(all(1:nrow(tmp) == tmp$`.order`))
+    tmp <- tmp[, !names(tmp) %in% c(".pageby", ".order")]
```

Comment:
To preserve the original dimensionality, use `drop = FALSE`. https://adv-r.hadley.nz/subsetting.html#simplify-preserve

Reply:
updated